infra/proposer.go

```diff
@@ -52,15 +52,17 @@ func (p *Proposer) Start(signed, processed chan *Elecments, done <-chan struct{}
 	for {
 		select {
 		case s := <-signed:
-			r, err := p.e.ProcessProposal(context.Background(), s.SignedProp)
-			if err != nil || r.Response.Status < 200 || r.Response.Status >= 400 {
-				fmt.Printf("Err processing proposal: %s, status: %d\n", err, r.Response.Status)
-				continue
+			endorsment := make([]*peer.ProposalResponse, len(p.e))
+			for n, e := range p.e {
+				r, err := e.ProcessProposal(context.Background(), s.SignedProp)
+				if err != nil || r.Response.Status < 200 || r.Response.Status >= 400 {
+					fmt.Printf("Err processing proposal: %s, status: %d\n", err, r.Response.Status)
+					continue
+				}
+				endorsment[n] = r
 			}
```

Comment:
this would be bottleneck if done serially. Let's parallelize it

Reply:
em... if we need both org1 and org2 endorsements,  before send to orderer. Will it be bottleneck to do it serially? 

in that case, we should not send to orderer until we collected for two endorsements.
if org1's peer is out of service.
No matter parallel or serially,  we can't get the endorsement.

if org1's peer or org2's peer is slow.
The totally TPS is decided by the slowest one. 
In parallel, there will be lots of single endorsement...

Reply:
Updated into parallel, ref http://www.golangpatterns.info/concurrency/semaphores